packages/shell-api/src/mongo.ts

```diff
@@ -365,6 +368,18 @@ export default class Mongo extends ShellApiClass {
     return await this._listDatabases(options);
   }
 
+  @returnsPromise
+  @serverVersions(['8.0.0', ServerVersions.latest])
+  @apiVersions([1])
+  bulkWrite(
+    models: AnyClientBulkWriteModel<Document>[],
+    options: ClientBulkWriteOptions = {}
+  ): Promise<ClientBulkWriteResult> {
```

Comment:
We've typically wrapped result types in our own types so that we're a bit more independent when it comes to server changes, see `result.ts` – I'd consider doing that here as well if we can (this mostly already matches our existing` BulkWriteResult` as well?)